main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -2238,7 +2238,12 @@ object Weeder extends Phase[ParsedAst.Program, WeededAst.Program] {
   /**
     * Weeds the given documentation.
     */
-  private def visitDoc(doc0: ParsedAst.Doc): Ast.Doc = Ast.Doc(doc0.lines.toList, mkSL(doc0.sp1, doc0.sp2))
+  private def visitDoc(doc0: ParsedAst.Doc): Ast.Doc = {
+    val trimmedLines = doc0.lines.map(_.trim)
+    val trimmedBeginning = trimmedLines.dropWhile(_ == "")
+    val trimmedEnd = trimmedBeginning.reverse.dropWhile(_ == "").reverse
```

Comment:
takeWhile?

Reply:
That would stop at the first empty line. I couldn't find a nice way other than this and my brief internet search found nothing better

Reply:
Ok, I understand.